ci/scripts/python_build.sh

```diff
@@ -70,6 +70,7 @@ export PYARROW_WITH_SUBSTRAIT=${ARROW_SUBSTRAIT:-OFF}
 export PYARROW_PARALLEL=${n_jobs}
 
 export LD_LIBRARY_PATH=${ARROW_HOME}/lib:${LD_LIBRARY_PATH}
+export CMAKE_PREFIX_PATH=${ARROW_HOME}:${CMAKE_PREFIX_PATH}
```

Comment:
Do we need this?
If we don't need this, I want to revert this before we merge this.

Reply:
that fixed one of the build errors, but I haven't tried without it since adding your fix. I'll try removing it and re-run the crossbow build, if all works fine then we're good.

Reply:
looks good with it reverted! all is passing. Thanks for the help!